Total Integration

edwin-pixel-1 opened the pull request:
Total integration with all features.
- Load splash display
- Show list subsribers with swipe refresh
- Show user details with option of open user github profile in browser
- Show list of user reporitories allowing each repo open in browser 

![demo1](https://cloud.githubusercontent.com/assets/929504/18215817/ccd35dae-7118-11e6-861c-677db6728464.gif)

elabcius5 committed:
CREATED A FRAGMENT CLASS

I created a Fragment class that contains a details of subriber seleted

elabcius5 committed:
RENAMED DE CLASS

edwin-pixel-1 committed:
ADDED COMMUNICATION BETWEEN FRAGMENTS

- Added UI layout to SubscriberDetails Fragment
- Added Click listener to SubscribersList Items
- Added communication between fragments and their transition.

edwin-pixel-1 committed:
Merge branch 'DetailsSubscriber-UI' into DetailsUser

edwin-pixel-1 committed:
Merge branch 'DetailsRepository' into DetailsUser

edwin-pixel-1 committed:
FIXED GRADLE

- Modified SDK version to 23 and each library.

elabcius5 committed:
DELETED REPEATED CLASSES

I deleted Adaptar and Item class beacuse these classes was repetead

elabcius5 committed:
MODIFIED ADAPTER TO SET ALL IN VIEWHOLDER

- Removed SubscriberItem class
- Modified SubscriberAdapter to set each view in ViewHolder class by
Edwin

edwin-pixel-1 committed:
MODIFIED REPOSITORIES ADAPTER

- Modified adapter of repositories to set all in the ViewHolder
- Added click listener to repositories recyclerView

elabcius5 committed:
MODIFIED REPOSITORIES ADAPTER

Modified adapter of repositories to set all in the ViewHolder. By Edwin

elabcius5 committed:
MERGED WITH MASTER

Merged branch with master and resolved the conflicts

edwin-pixel-1 committed:
ADDED FUNCTIONALITY TO OPEN LINKS IN BROWSER

- Open User Link in browser clicking on Full Name
- Open Repo link in browser clicking on Item list

elabcius5 committed:
CORRECTION BASED TO PAST REVISIONS

edwin-pixel-1 committed:
Merge branch 'DetailsUser-openLinks' into DetailsUser_BACK

edwin-pixel-1 committed:
Merge branch 'DetailsUser' into DetailsUser_BACK

edwin-pixel-1 committed:
Merge branch 'DetailsUser_BACK' of https://github.com/alejo740/GithubSubscribers into DetailsUser_BACK

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -0,0 +1,88 @@
+        @Override
+        public void onClick(View v) {
+            if (clickListener != null) {
+                clickListener.onClickItemList(this, getAdapterPosition());
```
You can just pass the selected item :smile:

daniellemariani replied:
+1

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -24,65 +27,81 @@
+        repositoriesAdapter = new RepositoryAdapter();
+    }
+
+    public static SubscriberDetailFragment newInstance(String param1) {
```
Rename param1 to a proper name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -24,65 +27,81 @@
-        txtFollowingCounter = (TextView) viewFragment.findViewById(R.id.txt_following_counter_subscriber_detail);
-        txtReposCounter = (TextView) viewFragment.findViewById(R.id.txt_repos_counter_subscriber_detail);
+
+        profileImage = (ImageView) viewFragment.findViewById(R.id.img_avatar_subscriber_detail);
```
Research about what we can use to avoid doing a `findViewById` for each view.

edwin-pixel-1 replied:
We think this could be improved with a dataBinding system. For example, **DataBinding** Library of Android or **butterknife**

lucaslabs replied:
Butterknife is a simpler and quicker approach :smiley:

edwin-pixel-1 replied:
https://trello.com/c/DH9kS2uk/17-butterknife-for-data-binding

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -0,0 +1,88 @@
+            repoUrl = item.getHtmlUrl();
+        }
+
+        public String getRepoUrl() {
```
This field belogns to the model/item not the view. Remove it.

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:
```diff
@@ -26,9 +26,8 @@ public SubscriberDetailPresenterImpl(SubscriberDetailView view) {
-    @Override
     public void onFinishedUser(User userItem) {
-        view.showSubscriberUser(userItem);
+            view.showSubscriberDetails(userItem);
```
apply formatter

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/commons/utils/Utils.java:
```diff
@@ -16,4 +19,10 @@ public static void debugLog(String msg) {
     }
 
+    public static void openLinkInBrowser(Context context, String url){
+        Intent i = new Intent(Intent.ACTION_VIEW);
```
Choose more descriptive names

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/commons/utils/Utils.java:
```diff
@@ -16,4 +19,10 @@ public static void debugLog(String msg) {
         }
     }
 
+    public static void openLinkInBrowser(Context context, String url){
```
Check code format here (Space is missing)

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -0,0 +1,88 @@
+        }
+    }
+
+    public interface ItemRepoClickListener {
```
Rename to `OnRepositoryItemClickListener` to follow android conventions

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -24,65 +27,81 @@
-    private TextView txtLocation;
-    private TextView txtFollowingCounter;
-    private TextView txtReposCounter;
+    private static final String ARG_USERNAME = "ARG_USERNAME";
```
Nice :+1:

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -24,65 +27,81 @@
-    private TextView txtFollowingCounter;
-    private TextView txtReposCounter;
+    private static final String ARG_USERNAME = "ARG_USERNAME";
+    private String userNameArg;
```
I think this variable is a litle confusing with the previous static one. Rename it to just `userName`

edwin-pixel-1 committed:
FIXED SUGGESTIONS

- Fixed Rename param1 to a proper name.
- Removed getRepoUrl method from viewHolder adapter
- Fixed some code format
- Renamed ItemRepoClickListener to OnRepositoryItemClickListener
- Renamed Intent variable in Utils class

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -67,22 +70,17 @@ public SubscriberDetailViewHolder(View itemView) {
-    public interface ItemRepoClickListener {
-        void onClickItemList(SubscriberDetailViewHolder view, int position);
+    public interface OnRepositoryItemClickListener {
+        void onClickItemList(SubscriberDetailViewHolder view);
```
Please pass the item not the view.

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersAdapter.java:
```diff
@@ -61,11 +65,27 @@ public SubscriberViewHolder(View itemView) {
+        }
+    }
+
+    public interface ItemClickListener {
```
In general, in our projects we have a rule to put interfaces for classes at the top, even above fields

juliocatano commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListView.java:
```diff
@@ -12,8 +12,6 @@
  */
 public interface SubscribersListView {
 
-    void navigateToUserDetail();
```
Is this ok? to be deleting the `navigateToUserDetail`

elabcius5 replied:
This was removed because the fragment is implementing the click listeners so this method is private into the fragment

edwin-pixel-1 committed:
CHANGED OBJECT AS PARAM IN onClickItemList

- Changed argument of onClickItemList in ViewHolder RepositoryAdapter
  Class

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -75,12 +71,12 @@ public void onBind(Repository item) {
 
     public interface OnRepositoryItemClickListener {
-        void onClickItemList(SubscriberDetailViewHolder view);
+        void onClickItemList(Repository view);
```
Rename param name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -131,9 +131,8 @@ public void showSubscriberUserRepositories(List<Repository> repositories) {
-    public void onClickItemList(RepositoryAdapter.SubscriberDetailViewHolder view) {
-        Repository repo = repositoriesAdapter.getRepository(view.getAdapterPosition());
-        String repoUrl = repo.getHtmlUrl();
+    public void onClickItemList(Repository view) {
```
Rename param name.

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -0,0 +1,82 @@
+        notifyDataSetChanged();
+    }
+
+    public class SubscriberDetailViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```
If this adapter is called RepositoryAdapter, it would be nice that the ViewHolder is called RepositoryViewHolder as well.

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailView.java:
```diff
@@ -8,14 +8,14 @@
  *
- * @author juan.herrera
- * @since 30/08/2016
+ * @author edwin.cobos
```
Why don't you keep both author?

elabcius5 committed:
Merge branch 'master' into DetailsUser_BACK

elabcius5 committed:
Merge branch 'DetailsUser_BACK' of https://github.com/alejo740/GithubSubscribers into DetailsUser_BACK

elabcius5 committed:
CORRECTION BASED IN REVISIONS AND COMMENTS

- Renamed the viewholder of repository class.
- Moved the interface at top of the SubscribersAdapter class.
- Added new author in the SubscriberDetailView class

edwin-pixel-1 committed:
Fixed ViewHolder on SubscribersAdapter class

- Fixed onClickItemList param to pass a Subscriber Object
- Moved ItemClickListener interface at top of SubscribersAdapter class

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -21,6 +21,10 @@
 public class RepositoryAdapter extends RecyclerView.Adapter<RepositoryAdapter.RepositoryViewHolder> {
 
+    public interface OnRepositoryItemClickListener {
+        void onClickItemList(Repository view);
```
Rename param with a proper name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersAdapter.java:
```diff
@@ -25,7 +25,7 @@
 
     public interface ItemClickListener {
-        void onClickItemList(SubscriberViewHolder view, int position);
+        void onClickItemList(Subscriber view);
```
Rename param with a proper name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListFragment.java:
```diff
@@ -72,8 +72,9 @@ public void onDetach() {
     @Override
-    public void onClickItemList(SubscribersAdapter.SubscriberViewHolder view, int position) {
-        mListener.onChangeToSubscriberDetails(view.getUserName());
+    public void onClickItemList(Subscriber view) {
```
Rename param with a proper name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/RepositoryAdapter.java:
```diff
@@ -0,0 +1,82 @@
+public class RepositoryAdapter extends RecyclerView.Adapter<RepositoryAdapter.RepositoryViewHolder> {
+
+    public interface OnRepositoryItemClickListener {
+        void onClickItemList(Repository view);
```
Rename param with a proper name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -93,4 +112,34 @@ public void showRepositoryError() {
+    }
+
+    @Override
+    public void onClickItemList(Repository view) {
```
Rename param with a proper name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersAdapter.java:
```diff
@@ -23,8 +24,12 @@
 
-    private List<Subscriber> subscriberList;
+    public interface ItemClickListener {
+        void onClickItemList(Subscriber view);
```
Rename param with a proper name.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListFragment.java:
```diff
@@ -48,10 +50,34 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+    }
+
+    @Override
+    public void onClickItemList(Subscriber view) {
```
Rename param with a proper name.

edwin-pixel-1 committed:
Renamed param of Clikc Listener method

- Renamed param of onClickItemList method in interface RepositoryAdapter.OnRepositoryItemClickListener

- Renamed param of onClickItemList method in interface SubscribersAdapter.ItemClickListener

lucaslabs commented:
LG

daniellemariani commented:
LG
